src/v/cluster/members_table.cc

```diff
@@ -158,20 +157,19 @@ members_table::apply(model::offset version, maintenance_mode_cmd cmd) {
     if (target == _brokers.end()) {
         return errc::node_does_not_exists;
     }
+    auto& [id, broker] = *target;
 
     // no rules to enforce when disabling maintenance mode
     const auto enable = cmd.value;
     if (!enable) {
         if (
-          target->second->get_maintenance_state()
+          broker->get_maintenance_state()
           == model::maintenance_state::inactive) {
             return errc::success;
         }
 
-        target->second->set_maintenance_state(
-          model::maintenance_state::inactive);
-        notify_maintenance_state_change(
-          target->second->id(), model::maintenance_state::inactive);
+        broker->set_maintenance_state(model::maintenance_state::inactive);
+        notify_maintenance_state_change(id, model::maintenance_state::inactive);
```

Comment:
`id` vs `target->second->id(),`
that one isn't technically replacing `target->second` with `target->first`. i assume they are equal tho?

seems like something that would be useful to call out in the commit message so reviewer knows that it was as aspect that was considered.

Reply:
Yeah good point, it's not obvious for reviewers. Updated the commit message.

They are identical:

```
_brokers.emplace(br.id(), ss::make_lw_shared<model::broker>(br));
```